Lib/test/test_tools/test_msgfmt.py

```diff
@@ -43,6 +43,7 @@ def test_compilation(self):
                         actual = GNUTranslations(f)
 
                     self.assertDictEqual(actual._catalog, expected._catalog)
+                    self.assertFalse(msgfmt.MESSAGES)
```

Comment:
The compilation here runs in a subprocess so you won't be able to observe the global variable from the parent process.